cmd/sys-mgmt-executor/main.go

```diff
@@ -0,0 +1,212 @@
+
+	var service = ""
+	var operation = ""
+
+	if len(os.Args) > 2 {
+		service = os.Args[1]
+		operation = os.Args[2]
+
+		switch operation {
+		case START:
+			err := ExecuteDockerCommands(service, "start")
+			if err != nil {
+				fmt.Println("error restarting service: ", service)
+			} else {
+				fmt.Println("success in restarting service: ", service)
+			}
+			break
+
+		case STOP:
+			err := ExecuteDockerCommands(service, "stop")
+			if err != nil {
+				fmt.Println("error stopping service: ", service)
+			} else {
+				fmt.Println("success in stopping service: ", service)
+			}
+			break
+
+		case RESTART:
+			err := ExecuteDockerCommands(service, "restart")
+			if err != nil {
+				fmt.Println("error restarting service: ", service)
+			} else {
+				fmt.Println("success in restarting service: ", service)
+			}
+			break
+
+		default:
+			fmt.Println("unknown operation was requested: ", operation)
+			break
+
+		}
+	}
+}
+
+func findDockerContainerStatus(service string, status string) bool {
+
+	var (
+		cmdOut []byte
+		err    error
+	)
+	cmdName := "docker"
+	cmdArgs := []string{"ps"}
+	if cmdOut, err = exec.Command(cmdName, cmdArgs...).CombinedOutput(); err != nil {
+		fmt.Println("error running the docker command", "error message", err.Error())
+		os.Exit(1)
+	}
+
+	dockerOutput := string(cmdOut)
+
+	// Find whether the container to start has started.
+	for _, line := range strings.Split(strings.TrimSuffix(dockerOutput, "\n"), "\n") {
+		if strings.Contains(line, service) {
+
+			if status == "Up" {
+				if strings.Contains(line, "Up") {
+					fmt.Println("container started", "service name", service, "details", line)
+					return true
+				} else {
+					fmt.Println("container NOT started", "service name", service)
+					return false
+				}
+			} else if status == "Exited" {
+				if strings.Contains(line, "Exited") {
+					fmt.Println("container stopped", "service name", service, "details", line)
+					return true
+				} else {
+					fmt.Println("container NOT stopped", "service name", service)
+					return false
+				}
+			}
+		}
+	}
+	return false
+}
+
+func ExecuteDockerCommands(service string, operation string) error {
+	_, knownService := services[service]
+
+	if knownService {
+		runDockerCommands(service, services[service], operation)
+
+		return nil
+	} else {
+		newError := fmt.Errorf("unknown service: %v", service)
+		fmt.Println(newError.Error())
+
+		return newError
+	}
+}
+
+func runDockerCommands(service string, dockerService string, operation string) {
+
+	var (
+		err    error
+		cmdDir string
+	)
+
+	cmdName := "docker"
+
+	cmdArgs := []string{operation, dockerService}
+	cmd := exec.Command(cmdName, cmdArgs...)
+	cmd.Dir = cmdDir
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Println("docker command failed", "error message", err.Error())
+		fmt.Println("associated output", "error message", string(out))
+	} else if operation == "start" {
```

Comment:
This shouldn't be an `else if`. If the err != nil, print the error message and return. Subsequent to that handle the operation values via a switch. The handling of start/restart are identical so they should fall-through to the same logic with stop being its own case.

Reply:
Excellent catch @tsconn23. I have also replaced `fmt statements` (which were a carry-over from when this SMA docker-compose executor was conceived as located in a standalone repo,  not as part of edgex-go mono repo) by suitable LoggingClient statements.